actionpack/test/controller/render_test.rb

```diff
@@ -162,6 +162,17 @@ def conditional_hello_with_cache_control_headers
     render action: "hello_world"
   end
 
+  def conditional_hello_with_expires_and_confliciting_cache_control_headers
+    response.headers["Cache-Control"] = "no-cache, must-revalidate"
+    expires_now
```

Comment:
Also, the reason the `must-revalidate` is stripped, is because of [this bit of logic](https://github.com/rails/rails/blob/665ac7cff212d010a3573f85cea895666fbaad15/actionpack/lib/action_dispatch/http/cache.rb#L195-L199). And, since `must-revalidate` is considered [a special key](https://github.com/rails/rails/blob/665ac7cff212d010a3573f85cea895666fbaad15/actionpack/lib/action_dispatch/http/cache.rb#L129), it is not added back as an "extra". I wasn't 100% clear on the history here, but I can see how `no-cache` might invalidate the need for `must-revalidate`. That bit seems fine, but it is strange that we can set it in one case, but not in another (based on whether `response.cache_control` ends up getting touched).